### ISSUE TITLE
Proposal: link Preconnect for CDN 

### ISSUE THREAD

**Issue by keesspoelstra** [apache/incubator-pagespeed-mod#1869]
We see more people using CDNs for their static resources. The connect is pretty efficient and early in the processing of the page because resources appear early in the HTML. 
It can however be improved a little bit by hinting a preconnect in a HTTP link header.

Any thoughts on how to configure this? Manual or detect this in the page?

**Comment by jmarantz** [comment]
Does https://www.modpagespeed.com/doc/filter-insert-dns-prefetch do what you want?

**Comment by keesspoelstra** [comment]
No, preconnect will actually setup a connection to the server, because of the SSL/TLS handshake song and dance this is actually a substantial amount of time. 

I'm not entirely sure, but in some traces I saw preconnect has priority over dns-prefetch.
